### PR TITLE
Fix add node kubeadm

### PR DESCRIPTION
**Pull request by stgleb**
Save bootstrap token in kube
Restore that token when adding node

### CODE REVIEW

File: pkg/workflows/steps/amazon/create_machine.go [hunk 1]
```diff
@@ -144,16 +144,15 @@ func (s *StepCreateInstance) Run(ctx context.Context, w io.Writer, cfg *steps.Co
...
-		runInstanceInput.NetworkInterfaces = []*ec2.InstanceNetworkInterfaceSpecification{
-			{
-				DeviceIndex:              aws.Int64(0),
-				AssociatePublicIpAddress: aws.Bool(true),
-				DeleteOnTermination:      aws.Bool(true),
-				SubnetId:                 aws.String(cfg.AWSConfig.Subnets[cfg.AWSConfig.AvailabilityZone]),
-				Groups:                   []*string{secGroupID},
-			},
-		}
+
```
Comment (stgleb) [inline pkg/workflows/steps/amazon/create_machine.go]: every node has public IP address by default, so skip this if statement

File: templates/kubeadm.sh.tpl [hunk 2]
```diff
@@ -16,12 +16,12 @@ sudo systemctl restart kubelet
 sudo kubeadm config images pull
 
 {{ if .IsBootstrap }}
-sudo kubeadm init --ignore-preflight-errors=NumCPU --apiserver-advertise-address={{ .AdvertiseAddress }} --token={{ .Token }} --pod-network-cidr={{ .CIDR }} \
```
Comment (stgleb) [inline templates/kubeadm.sh.tpl]: By default token is valid for 24 hours, but we need token that allows to join nodes for longer period.

File: pkg/kube/handler.go [hunk 3]
```diff
@@ -531,15 +531,28 @@ func (h *Handler) addMachine(w http.ResponseWriter, r *http.Request) {
...
 		return
 	}
 
+	logrus.Debugf("Get cloud profile %s", k.ProfileID)
+	kubeProfile, err := h.profileSvc.Get(r.Context(), k.ProfileID)
+
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+		if sgerrors.IsNotFound(err) {
+			message.SendNotFound(w, k.ProfileID, err)
```
Comment (saheienko) [inline pkg/kube/handler.go]: This is kubes handler. Should it be status `Bad Request` if it doesn't found profile id?

Comment (stgleb) [reply pkg/kube/handler.go]: but 404 give better clue on what happens. 400 bad request is mostly associated with bad formatted data.

**Pull request merged**